homeassistant/components/holiday/config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Config flow for Holiday integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from holidays import country_holidays, list_supported_countries
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_COUNTRY, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    CountrySelector,
+    CountrySelectorConfig,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+)
+
+from .const import CONF_PROVINCE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): TextSelector(),
+        vol.Required(CONF_COUNTRY): CountrySelector(
+            CountrySelectorConfig(
+                countries=list(list_supported_countries(include_aliases=False)),
+            )
+        ),
+    }
+)
+
+
+class HolidayConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Holiday."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.data = user_input
+
+            if self.data[CONF_COUNTRY]:
```

Comment:
When is country not set given that it is required? Seems like this is always true so the if statement can be omitted.

Reply:
I got some mypy errors in between, so I added that. I'll remove it.